src/client/game-over.tsx

```diff
@@ -0,0 +1,85 @@
+import * as React from 'react';
+
+interface GameOverProps {
+  isWinner: boolean;
+  maxCountdownTime: number;
+  // onPlayAgain: () => void;
+  onBackToMainMenu: () => void;
+}
+interface GameOverState {
+  countdownTime: number;
+}
+
+export class GameOver extends React.Component<GameOverProps, GameOverState> {
+  countdownTimer: number;
+
+  constructor(props: GameOverProps) {
+    super(props);
+
+    //not really necessary for just going back to main menu, but would be if rematch was an option
+    this.state = {countdownTime: this.props.maxCountdownTime};
+    this.countdownTimer = setTimeout(this.updateCountdown.bind(this), 1000, this.state.countdownTime - 1);
+  }
+
+  updateCountdown(time: number) {
+    this.setState({countdownTime: time});
```

Comment:
If you're storing `countdownTime` in state, no need to also store it in the function parameters and in GameOver.
just do
```js
this.setState({countdownTime: this.state.countdownTime - 1}, () => {
  if (this.state.countdownTime - 1 >= 0) {  setTimeout(this.updateCountdown.bind(this), 1000); }
  else { this.props.onBackToMainMenu() }
});